crates/oxc_linter/src/rules/import/max_dependencies.rs

```diff
@@ -0,0 +1,202 @@
+        Self { max: 10, ignore_type_imports: false }
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Forbid modules to have too many dependencies (import or require statements).
+    ///
+    /// ### Why is this bad?
+    ///
+    /// This is a useful rule because a module with too many dependencies is a code smell, and
+    /// usually indicates the module is doing too much and/or should be broken up into smaller
+    /// modules.
+    ///
+    /// ### Example
+    ///
+    /// Given `{"max": 2}`
+    /// ```javascript
+    /// import a from './a';
+    /// import b from './b';
+    /// import c from './c';
+    /// ```
+
+    MaxDependencies,
+    pedantic,
+);
+
+impl Rule for MaxDependencies {
+    fn from_configuration(value: Value) -> Self {
+        let config = value.get(0);
+        if let Some(max) = config
+            .and_then(Value::as_number)
+            .and_then(serde_json::Number::as_u64)
+            .and_then(|v| usize::try_from(v).ok())
+        {
+            Self(Box::new(MaxDependenciesConfig { max, ignore_type_imports: false }))
+        } else {
+            let max = config
+                .and_then(|config| config.get("max"))
+                .and_then(Value::as_number)
+                .and_then(serde_json::Number::as_u64)
+                .map_or(10, |v| usize::try_from(v).unwrap_or(10));
+            let ignore_type_imports = config
+                .and_then(|config| config.get("ignoreTypeImports"))
+                .and_then(Value::as_bool)
+                .unwrap_or(false);
+
+            Self(Box::new(MaxDependenciesConfig { max, ignore_type_imports }))
+        }
+    }
+
+    fn run_once(&self, ctx: &LintContext<'_>) {
+        let module_record = ctx.module_record();
+        let mut module_count = module_record.import_entries.len();
+
+        if self.ignore_type_imports {
+            let type_imports =
+                module_record.import_entries.iter().filter(|entry| entry.is_type).count();
+
+            module_count -= type_imports;
+        }
+
+        if module_count <= self.max {
+            return;
+        }
+
+        let Some(entry) = module_record.import_entries.get(self.max) else {
+            return;
+        };
+
+        let error = format!(
+            "File has too many dependencies ({}). Maximum allowed is {}.",
+            module_count, self.max,
+        );
+        ctx.diagnostic(max_dependencies_diagnostic(&error, entry.module_request.span()));
+    }
+}
+
+#[test]
+fn test() {
+    use serde_json::json;
+
+    use crate::tester::Tester;
+
+    let pass = vec![
+        (
+            r"
+            import './foo.js';
+            ",
+            None,
+        ),
+        (
+            r"
+             import './foo.js';
+             import './bar.js';",
+            None,
+        ),
+        // (
+        //     r"
+        //      import './foo.js';
+        //      import './bar.js';
+        //      const a = require('./foo.js');
```

Comment:
Do you intend on adding CJS support?

Reply:
I don't, other rules also do not support CJS nor does the `module_record` know about them.

I also saw this rule ignore `require()`.
https://github.com/oxc-project/oxc/blob/09bc0fba23cd9ef9ce59e21e94f2100ef41b7b2e/crates/oxc_linter/src/rules/import/named.rs#L168